elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCleanerThread.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.core.filter.dialect.RSQLFilterDialect;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.request.EntityProjection;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Runnable thread for updating AsyncQueryThread status
+ * beyond the max run time and if not terminated by interrupt process
+ * due to app/host crash or restart.
+ */
+@Slf4j
+public class AsyncQueryCleanerThread implements Runnable {
```

Comment:
To be on the safe side, I would add lombok hashcode and equals to all threads that are being managed by a thread pool.

Reply:
We are already adding `@Data` for AsyncQueryCleanerThread.java which internally includes `@EqualsAndHashCode`. Do we need to add these separately for the threads?

Reply:
`@Data` should be sufficient.